code/modules/mob/living/carbon/xenomorph/castes/larva/larva.dm

```diff
@@ -56,9 +56,9 @@
 	var/progress = "" //Naming convention, three different names
 
 	switch(amount_grown)
-		if(0 to 49) //We're still bloody
+		if(0 to 5) //We're still bloody
 			progress = "Bloody "
-		if(100 to INFINITY)
+		if(6 to INFINITY)
 			progress = "Mature "
```

Comment:
Instead of `amount_grown` with hardcoded values, do a new ratio variable, something like this:
`var/growth_progress = (amount_grown * 100) / max_grown`
Then you can just do a the switch using the percentage, and if you change the values it will be fine.